src/url-state-machine.js

```diff
@@ -43,6 +43,41 @@ function isASCIIHex(c) {
   return isASCIIDigit(c) || (c >= 0x41 && c <= 0x46) || (c >= 0x61 && c <= 0x66);
 }
 
+function isURLCodePoint(c) {
+  return (
+    isASCIIAlphanumeric(c) ||
+      c === 0x21 ||
```

Comment:
Can you switch these over to use the notation used in the URL spec (that is, use the literal chars, pass them through our conversion `p` transform). Optionally transform it to an array (not sure if that makes copying them off the spec easier)?